lightpath/tests/test_gui.py

```diff
@@ -68,15 +69,15 @@ def test_upstream_check(lightapp: LightApp, monkeypatch):
             row[0].setHidden.assert_called_with(False)
 
 
-def test_filtering(lightapp: LightApp, monkeypatch):
+def test_filtering(qtbot: QtBot, lightapp: LightApp, monkeypatch):
     lightapp.destination_combo.setCurrentIndex(4)  # set current to MEC
     # Create mock functions
     for row in lightapp.rows:
         monkeypatch.setattr(row[0], 'setHidden', Mock())
     # Initialize properly with nothing hidden
     lightapp.filter()
     for row in lightapp.rows:
-        row[0].setHidden.assert_called_with(False)
+        qtbot.waitUntil(lambda: row[0].setHidden.assert_called_with(False))
```

Comment:
How do these statements work? The encapsuled `assert_called_with` is actually an assert statement. Does it have some undocumented return value?
https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_with

Reply:
Nevermind, I didn't know qtbot would wait until the assertion passed

Reply:
Yea this is some fanciness I picked up while filling out the atef test suite.